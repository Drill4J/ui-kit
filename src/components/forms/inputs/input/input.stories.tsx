import React from 'react';
import { ComponentMeta, Story } from '@storybook/react';
import 'twin.macro';

import { Input } from './input';

export default {
  title: 'Input',
  component: Input,
} as ComponentMeta<typeof Input>;

const Template: Story = (args) => <Input {...args} />;
export const Default = Template.bind({});
Default.args = { disabled: false };
Default.parameters = {
  docs: {
    source: {
      code: '<Input />',
    },
  },
};

export const Disabled = Template.bind({});
Disabled.args = { disabled: true };
Disabled.parameters = {
  docs: {
    source: {
      code: '<Input disabled/>',
    },
  },
};
