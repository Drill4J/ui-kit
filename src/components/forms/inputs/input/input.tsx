import tw, { styled, css } from 'twin.macro';

export const Input = styled.input<{disabled?: boolean; error?: boolean}>`
  ${tw`font-regular`};
  box-sizing: border-box;
  width: 100%;
  height: 40px;
  padding: 0 16px;
  line-height: 22px;
  border: 1px solid #e3e6e8;
  border-radius: 4px;
  background-color: #ffffff;
  color: #1b191b;
  font-size: 14px;
  outline: none;

  &:focus {
    border: 1px solid #1b191b;
  }

  &::placeholder {
    color: #687481;
  }

  ${({ disabled, error }) => [
    disabled && css`
      border: 1px solid #e3e6e8;
      background-color: #f8f9fb;
      color: #687481;
    `,
    error && css`border: 1px solid #ee0000`,
  ]}
`;
