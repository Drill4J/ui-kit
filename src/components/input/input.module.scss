@import '../../common/style-guide/index';

.input-wrapper {
  position: relative;

  &__input-element {
    width: 100%;
    height: $common-element-height;
    font-size: 16px;
    line-height: 22px;
    border: 1px solid $slate-gray;
    background-color: map-get($color-palette, white);
    padding: 0 48px;
    outline: none;

    &::placeholder {
      opacity: 0.5;
    }

    &--rounded {
      border-radius: 20px;
    }

    &--disabled {
      border: 1px solid map-get($color-palette, light-blue);
      background-color: $athens-gray;
    }
  }

  &__icon-wrapper {
    display: flex;
    justify-content: center;
    height: 20px;
    width: 20px;
    position: absolute;
    color: $slate-gray;
    left: 12px;
    top: 10px;
  }
}
