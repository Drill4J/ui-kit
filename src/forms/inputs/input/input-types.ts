export interface InputProps {
  className?: string;
  icon?: React.ReactNode;
  placeholder?: string;
  rounded?: boolean;
  disabled?: boolean;
  type?: string;
  checked?: boolean;
  value?: any;
  onChange?: () => void;
  error?: boolean;
  success?: boolean;
  ref?: React.Ref<HTMLInputElement>;
  prefix?: boolean;
}
