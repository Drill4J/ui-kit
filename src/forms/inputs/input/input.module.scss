@import '../../../style-guide/index';

.input-wrapper {
  position: relative;

  &__input-element {
    box-sizing: border-box;
    width: 100%;
    height: $common-element-height;
    padding: 0 16px;
    line-height: 22px;
    border: 1px solid map-get($monochrome, medium-tint);
    border-radius: 4px;
    background-color: map-get($monochrome, white);
    outline: none;
    font-family: $regular-font;
    font-size: 14px;
    color: map-get($monochrome, black);

    &:focus {
      border: 1px solid map-get($monochrome, black);
    }

    &::placeholder {
      opacity: 0.5;
    }

    &--rounded {
      border-radius: 20px;
    }

    &--disabled {
      border: 1px solid map-get($monochrome, medium-tint);
      background-color: map-get($monochrome, light-tint);
      color: map-get($monochrome, black);
      opacity: 0.5;
    }

    &--success {
      border: 1px solid map-get($green, default);
    }

    &--error {
      border: 1px solid map-get($red, default);
    }

    &--prefix {
      padding: 0 40px;
    }
  }

  &__icon-wrapper {
    display: flex;
    justify-content: center;
    height: 20px;
    width: 20px;
    position: absolute;
    color: map-get($monochrome, default);
    left: 12px;
    top: 10px;
  }
}
