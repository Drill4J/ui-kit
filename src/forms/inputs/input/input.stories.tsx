import React from 'react';
import { storiesOf } from '@storybook/react';

import { Input } from './input';
import { Icons } from '../../../components/icon';

storiesOf('Input', module).add('Input types', () => (
  <>
    <h3>Simple input</h3>
    <Input
      icon={<Icons.Account />}
      placeholder="enter text..."
      prefix
    />
    <h3>Rounded input</h3>
    <Input
      placeholder="enter text..."
      rounded
    />
    <h3>Disabled input</h3>
    <Input
      placeholder="enter text..."
      disabled
    />
    <h3>Success input</h3>
    <Input
      placeholder="enter text..."
      success
    />
    <h3>Error input</h3>
    <Input
      placeholder="enter text..."
      error
    />
  </>
));
