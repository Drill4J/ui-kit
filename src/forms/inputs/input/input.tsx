import * as React from 'react';
import { BEM, input as bemInput } from '@redneckz/react-bem-helper';

import { INPUT_PROPS } from './input-props';
import { InputProps } from '../input-types';

import styles from './input.module.scss';

const input = BEM(styles);

const INPUT_PROPS_OBJ = Object.assign({}, ...INPUT_PROPS.map((key) => ({ [key]: undefined })));

export const Input = input(({
  className, ...restProps
}: InputProps) => (
  <div className={className}>
    <InputElement {...restProps} />
  </div>
));

const InputElement = input.inputElement(
  bemInput({
    ...INPUT_PROPS_OBJ,
    value: '',
  } as {}),
);
