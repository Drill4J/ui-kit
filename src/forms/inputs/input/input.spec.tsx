import React from 'react';
import renderer from 'react-test-renderer';
import { mount } from 'enzyme';

import { Input } from './input';

describe('Input', () => {
  const mockFn = jest.fn();
  const props = {
    onChange: mockFn,
    rounded: false,
    disabled: false,
    checked: false,
    error: false,
    placeholder: 'placeholder initial state',
    value: 'input component',
    type: 'text',
  };

  it('Should render a Input', () => {
    const tree = renderer
      .create(
        <>
          <Input
            placeholder="enter text..."
          />
          <Input
            placeholder="enter text..."
            rounded
          />
          <Input
            placeholder="enter text..."
            disabled
          />
          <Input
            placeholder="enter text..."
            success
          />
          <Input
            placeholder="enter text..."
            error
          />
        </>,
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('should render component with initial state', () => {
    const wrapper = mount(
      <Input {...props} />,
    );
    wrapper.find('input').simulate('change');
    expect(mockFn).toHaveBeenCalled();
    expect(wrapper.prop('value')).toContain('input component');
    expect(wrapper.prop('placeholder')).toContain('placeholder initial state');
  });
});
