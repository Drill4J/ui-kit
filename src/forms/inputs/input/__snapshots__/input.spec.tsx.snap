// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input Should render a Input 1`] = `
Array [
  <div
    className="input-wrapper"
  >
    <input
      className="input-wrapper__input-element"
      placeholder="enter text..."
      value=""
    />
  </div>,
  <div
    className="input-wrapper input-wrapper--rounded"
  >
    <input
      className="input-wrapper__input-element input-wrapper__input-element--rounded"
      placeholder="enter text..."
      value=""
    />
  </div>,
  <div
    className="input-wrapper input-wrapper--disabled"
  >
    <input
      className="input-wrapper__input-element input-wrapper__input-element--disabled"
      disabled={true}
      placeholder="enter text..."
      value=""
    />
  </div>,
  <div
    className="input-wrapper input-wrapper--success"
  >
    <input
      className="input-wrapper__input-element input-wrapper__input-element--success"
      placeholder="enter text..."
      value=""
    />
  </div>,
  <div
    className="input-wrapper input-wrapper--error"
  >
    <input
      className="input-wrapper__input-element input-wrapper__input-element--error"
      placeholder="enter text..."
      value=""
    />
  </div>,
]
`;
